python/paddle/fluid/io.py

```diff
@@ -454,3 +452,167 @@ def get_parameter_value_by_name(name, executor, program=None):
         program = default_main_program()
     var = program.global_block().var(name)
     return get_parameter_value(var, executor)
+
+
+SUCCESS_MARK_FILENAME = "_SUCCESS"
+
+
+def save_checkpoint(executor,
+                    dirname=None,
+                    max_num_checkpoints=3,
+                    save_interval_secs=600,
+                    main_program=None):
+    """
+    Save Checkpoint will save persistable LodTensor variables from main_program in checkpoint directory,
+    directory named by serial number from 0 to (n -1), save_checkpoint use LRU strategy
+    to keep numbers of checkpoint directory,  the numbers of checkpoint directory are max_num_checkpoints at most,
+    The interval time between two save_checkpoint must great than or equal to save_interval_secs.
+    
+    :param dirname
+    :param max_num_checkpoints
+    :param save_secs
+    :param main_program
+    """
+    if dirname is None:
+        dirname = os.getcwd()
+
+    if not os.path.isdir(dirname):
+        os.makedirs(dirname)
+
+    serial = _get_lastest_checkpoint_dir(dirname)
+    if serial >= 0 and not _interval_secs_exceed(
+            os.path.join(dirname, str(serial)), save_interval_secs):
+        return
+
+    serial = serial + 1
+    cur_dir = os.path.join(dirname, str(serial))
```

Comment:
The checkpoint directories will be saved in "1", "2", etc. which may not make sense to users, I think it's better to be like "checkpoint_1", "checkpoint_2", and the "_SUCCESS" file can save a timestamp when the checkpoint is saved.